archiver/helpers.py

```diff
@@ -140,14 +157,25 @@ def hash_files_and_check_symlinks(source_path, abs_paths, max_workers=1, integri
             in zip(file_list, hashes_list)]
 
 
-def get_files_in_folder(folder_path):
-    file_list = []
-    for root, _, files in os.walk(folder_path):
+def get_files_in_folder(folder_path, include_dirs=False):
+    listing = []
+    for root, dirs, files in os.walk(folder_path):
         root_path = Path(root)
         for file in files:
             abs_file = root_path.joinpath(file)
-            file_list.append(abs_file)
-    return file_list
+            listing.append(abs_file)
+        dirs_ = []
+        for dir in dirs:
+            abs_dir = root_path.joinpath(dir)
+            if Path(abs_dir).is_symlink():
+                listing.append(abs_dir)
+            else:
+                if include_dirs:
+                    listing.append(abs_dir)
+                dirs_.append(dir)
+        dirs = dirs_
```

Comment:
This should be `dirs[:] = dirs_` to properly mutate the list and not just reassigning to a new list, which would be ignored by os.walk

Reply:
Ah, good catch. I assume my tests have been too simple :) Will improve